### PR TITLE
zmarkdown: fix image & one smiley

### PR DESCRIPTION
**Pull request by artragis**
Since v27 deployment we got some feedback, this PR fixes two of them :

- we cannot use `:P` anymore
- when we use image figures are not created. (fix #420)

### CODE REVIEW

File: packages/zmarkdown/config/remark.js [hunk 1]
```diff
@@ -140,6 +140,7 @@ const remarkConfig = {
       '^^': '/static/smileys/hihi.png',
       ':o': '/static/smileys/huh.png',
       ':p': '/static/smileys/langue.png',
+      ':P': '/static/smileys/langue.png',
```
[the remark this answers is not in the document]
Comment (artragis) [reply packages/zmarkdown/config/remark.js]: perhaps. What do you think @vhf

[the remark this answers is not in the document]
Comment (vhf) [reply packages/zmarkdown/config/remark.js]: I usually don't consider smileys to be case sensitive.

[the remark this answers is not in the document]
Comment (artragis) [reply packages/zmarkdown/config/remark.js]: ok, so I will modify remark-emoticons.

**Comment by artragis** [comment]
I think this is mergeable @vhf?  @davbaumgartner

**Comment by artragis** [comment]
@vhf: WHat I see here:

- PR title is OK
- tests are complete : block image, inline image, transposition in html & latex
- tests are what we expect : block image gives caption when alt title is given
- snapshots are consistent.

I don't see any point.

**Comment by artragis** [comment]
so @vhf what can we do? Those bugs are annoying for zds members et I don't understand what you require me to do.

**Comment by vhf** [comment]
What would be really annoying is adding more bugs. Please add tests dedicated to the case insensitive smileys.

**Comment by artragis** [comment]
Like this one? https://github.com/zestedesavoir/zmarkdown/pull/241/commits/9e37b2f2c7897f8e45674d7ac7cb35cf5ab532cf#diff-37df776abe73b7b3095b1f81808ffe4eR79

**Comment by vhf** [comment]
Yes, a bit like this one except:

- Without bugs (as I said on many of your previous PRs, you need to read the git diff before committing, this way you can check that what you commit is what you actually want to commit and you don't end up pushing broken things),
- In its own test case (as we do whenever we discover a bug: we add a test case to make sure there will be no regression, and we name this test case with whatever it is actually testing).

**Comment by artragis** [comment]
> Without bugs (as I said on many of your previous PRs, you need to read the git diff before committing, this way you can check that what you commit is what you actually want to commit and you don't end up pushing broken things),

As I said in the previous PRs, I DO read the diff.

But as I said in previous PRs snapshot are really hard to read. When you see a bug it's far more easier to tell me you've seen it.

AS this pr is little, I managed to get your point after reading it 5 times with focus on the snapshot. If you found out the error at the first glance, I really don't get the point of not telling it.

I will spli the test to extract the `:p versus :P`

**Comment by artragis** [comment]
> Regarding your other commit, wrapping an image in a figure, I'd like to understand how it should behave before reviewing the code. The two test cases are not enough to get the intent or to check whether this will behave as expected.

if you have an image which is not inline and not included inside a figure you must transform it as a figure. 

For example:

```
Hello world

![legend](image)
```

becomes

```html
<p>Hello world</p>
<figure>
<img src="image" alt=""/>
<figcaption>legend</figcaption>
</figure>
```
Nota : the blank "alt" is not mandatory, it is an old behaviour which was not the best (alt text should always be here)

If you have a legend, you get : 

```
Hello

![alt](url)
FIgure: Legend
```

```html
<p>Hello world</p>
<figure>
<img src="image" alt="alt"/>
<figcaption>Legend</figcaption>
</figure>
```

This case is the one treated by remark-captions

THen for inline images (emoticons and all images that has type `inlineImage`, nothing must change.

**Comment by artragis** [comment]
I would say "C" : 

```html
<figure><img src="http://bar" alt="foo"><figcaption>foo</figcaption></figure>

<figure><img src="http://bar"><figcaption>foo</figcaption></figure>
```

I mean : remark-captions doesn't wrap figures between `<p>` and `</p>`

**Comment by vhf** [comment]
> I mean : remark-captions doesn't wrap figures between <p> and </p>

It does. I don't think HTML allows `figure`s to be in `p`s though.

**Comment by artragis** [comment]
When I use figures in zds website I have no wrapping p

no_wrapping

**Comment by vhf** [comment]
You are looking at what your browser parsed. Your browser respects the HTML spec. You should look at the HTML sent by the webserver to your browser.

**Comment by artragis** [comment]
True engough, thanks for the tip.

**Comment by vhf** [comment]
See the `<p></p>` right above `<figure>` on your screenshot? That's your browser immediately closing the `<p>` opened here: `<p><figure>` because figure is not allowed in p. Seeing this made me think figure might not be allowed in p, that's why I checked and reported this above.

**Pull request closed without merging**